setup.py

```diff
@@ -20,6 +20,7 @@
         'cryptography>=1.7.2',
         'pyOpenSSL>=16.2.0',
         'msgpack-python>=0.4.2',
+        'python-magic'
```

Comment:
def not going to make python-magic a core synapse dep ( we still have to run on things other than linux ) but you could use the dyndeps helpers within your module to load it dynamically and exception if it's needed and not found.